### ISSUE TITLE
The "\n" is not rendered as a line break for subtitles

### ISSUE THREAD

**Issue by fieraoh** [Rafostar/clapper#301]
Fedora 36
Gnome 42.3
Wayland

![screenshot](https://user-images.githubusercontent.com/72320316/179437022-0ea369bc-d57e-4346-b11d-5135b6953b11.png)

**Comment by Rafostar** [comment]
@fieraoh 
Since you closed this, I would like to ask if this was either fixed upstream or did you open an issue about it there. If so, I would like to link it here for anyone else who might have similar problem. Thanks.

**Comment by fieraoh** [comment]
Oh no, looks like I have closed the wrong issue.

**Comment by Rafostar** [comment]
@fieraoh 
I am keeping this open only because I do not want problems like this to be lost and forgotten. As I explained some time ago, there is nothing actionable here about this. This issue should be probably opened on GStreamer gitlab as `subparse` element issue. Please consider reporting it there, so we can close this one here.

**Comment by fieraoh** [comment]
To be honest with you, from a technical standpoint, I can't maintain an issue there as I haven't the faintest idea what a "`subparse`" element issue is.

I turned a blind eye to it because the fix is pretty simple and  99% of subtitles don't use "\n" for inserting line breaks anyway.

For the sake of universal compatibility I'd entitle you to opening the issue on gstreamer yourself.

**Comment by Ferenc-** [comment]
`subparse` is the gstreamer plug-in responsible for parsing `.srt`, `.sub`, `.mpsub`, `.mdvd`, `.smi`, `.txt`, `.dks` and `.vtt` files.
And the issue is perhaps with [gstsubparseelement](https://gitlab.freedesktop.org/gstreamer/gstreamer/-/blob/main/subprojects/gst-plugins-base/gst/subparse/gstsubparseelement.c)?


I am guessing, that for a possible bugreport either a subfile would be needed which can reproduce the issue, or some logs which reveal what is going on.  
A wild idea would be running clapper like this:

````bash
export GST_DEBUG='*:6' && clapper ./YOUR_SUBTITLED_FILE  2>&1 | grep -iE 'subparseelement'
````
This should print something like:
````
0:00:00.426762063   643 0x7fd8ac09f980 LOG                 subparse gstsubparseelement.c:134:gst_sub_parse_data_format_autodetect: SubRip (time based) format detected
0:00:00.426768592   643 0x7fd8ac09f980 DEBUG               subparse gstsubparseelement.c:356:gst_sub_parse_type_find: SubRip format detected
````

Or if that doesn't happen then increase the scope a bit:
````bash
export GST_DEBUG='*:6' && clapper ./YOUR_SUBTITLED_FILE  2>&1 | grep -iE '(subparse|subtitle)'
````
 and providing the output. Parhaps that catches something.

**Comment by Rafostar** [comment]
Printing everything + grep will make things very slow. The recommended way of running above would probably be:

```sh
GST_DEBUG=*subparse*:6 clapper ./YOUR_SUBTITLED_FILE
```

But I am not sure if this is a bug. Looks more like this is simply not implemented, thus text is displayed as-is.

**Comment by windwaker87** [comment]
I've noticed that Clapper does not render `{\anX}` anchored lines in SRT subtitles. I know it's not part of the SRT spec, but players like `VLC` or `Celluloid` do support that.

Given the fact that `GStreamer` does the parsing this might also fall under the same category.

I'm just interested if it could be added as most of my video archive have SRT subtitles with anchoring as default (instead of SSA/ASS) for compatibility with TVs and other devices.

**Comment by Rafostar** [comment]
> Given the fact that GStreamer does the parsing this might also fall under the same category.

Yes. As explained in my first comment [here](https://github.com/Rafostar/clapper/issues/301#issuecomment-1187100225), `Clapper` does not do subtitles parsing but relies on our backend `GStreamer` for that. Not fixable here, please report on their gitlab.
